cloud-prepare/vpc-peering.md

```diff
@@ -0,0 +1,210 @@
+# Submariner VPC Peering Support
+
+This document describes the technical details and impact of our proposal design
+to include Virtual Private Cloud (VPC) Peering support for Submariner and
+`subctl` located at this [Pull Request](https://github.com/submariner-io/cloud-prepare/pull/190).
+
+## Abstract
+This idea was born meanwhile the CockroachDB project focused to deploy a
+multi-cluster setup of CockroachDB and communicate the different clusters across
+regions using Submariner.
+
+One of the requirements generated from that project was to deploy CockroachDB on
+AWS in different regions at the same time, using VPC Peering and configure VXLan
+(instead of IPsec) to increase network performance because VPC Peering
+connections are already secure and encrypted.
+
+In order to achieve that, we observed neither the Operator nor the `subctl` CLI
+supported VPC Peering setup, configuration or deployment. For that reason, we
+decided to provide such functionality to cloud-prepare and subctl.  With a set
+of sub-commands in subctl, developers will automatically provision and configure
+a VPC Peering between two clusters of the same cloud provider. Once
+cloud-prepare allows VPC Peering, the Submariner Operator will include the VPC
+Peering configuration as well.
+
+## Design
+
+The following sections include the technical aspects of this feature adding and
+how it works.
+
+### New `subctl` sub-commands
+
+We purpose to add some subcommands to `subctl` to use the new VPC Peering
+features described on this document.
+
+#### VPC Peering Subcommands
+
+This will be how new added sub-commands would be like.
+
+```bash
+$ ./bin/subctl cloud VPC Peering
+This command creates a VPC Peering between different clusters of the same cloud
+provider.
+
+Usage:
+  subctl cloud VPC Peering [command]
```

Comment:
See my above comment on the VPC Peering term. It would make sense for AWS and GCP, but not for Azure.